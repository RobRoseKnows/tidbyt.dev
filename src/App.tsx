import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route, Routes } from "react-router-dom";

import Main from './main/Main';
import store from './store';
import AppTheme from './theme/AppTheme';
import Home from './home/Home';

const ROUTES = require.context('../docs', true, /\.(md)$/)

export default function App() {

    function importAll(r) {
        let docs = [];
        ROUTES.keys().forEach((item) => {
            let doc = item.replace('.md', '').replace("./", '');
            let path = `/docs/${doc}`;
            docs.push(<Route key={path} path={path} element={<Main source={r(item).default} />} />);
        });
        return docs;
    }

    return (
        <Provider store={store} >
            <AppTheme>
                <BrowserRouter>
                    <Routes>
                        <Route path="/" element={<Home />} />
                        {importAll(ROUTES)}
                    </Routes>
                </BrowserRouter>
            </AppTheme>
        </Provider >
    )
}