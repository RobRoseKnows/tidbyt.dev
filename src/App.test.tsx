import * as React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

it('app loads without error', () => {
    const component = renderer.create(
        <App />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});