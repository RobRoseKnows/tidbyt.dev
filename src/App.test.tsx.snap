// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app loads without error 1`] = `
<div
  className="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-1401ke1-MuiTypography-root"
>
  Hello, App!
</div>
`;
